pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/ModeAggregationFunction.java

```diff
@@ -467,7 +501,11 @@ public ColumnDataType getFinalResultColumnType() {
   @Override
   public Double extractFinalResult(Map<? extends Number, Long> intermediateResult) {
     if (intermediateResult.isEmpty()) {
-      return DEFAULT_FINAL_RESULT;
+      if (_nullHandlingEnabled) {
```

Comment:
When `_nullHandlingEnabled` is `true`, and all input values are `null`, will this map be `null`?
Same for `merge()`

Reply:
_this map_ = `intermediateResult`?

In that case, no. `intermediateResult` is created in `extractAggregationResult(AggregationResultHolder)`, which calls `extractIntermediateResult` which returns not null.

Reply:
I'm adding a new test to show the result when all stored data is null